### PR TITLE
added scroll to prop sections of docs

### PR DESCRIPTION
**Pull request by kwongz**
### Checklist

- ~~For UI or styling changes, I have added a screenshot or gif showing before & after~~
- ~~I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)~~
- ~~I have added to the docs where applicable~~
- ~~I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

**Comment by kwongz** [comment]
Added <a href="#props-{name}" to doc's prop section. 

Click on Value, scrolls you to Value in prop section, as well as updates the url (/components/big-value/#props-value)

image
https://www.loom.com/share/6d859aa798c54eb2921146b1ad2900f2

Entering that URL in a new tab will scroll to the specified section once the page opens. However, an existing issue was discovered: In the example video, the Big Value component loads after the initial scroll. This change affects the height of the page, causing the viewport's scroll position to shift, so the intended scroll position is no longer accurate.

Test example for Current Docs site & Local branch with prop scroll
https://www.loom.com/share/233bc359290f4d61a533576265d8ffcb

**Comment by kwongz** [comment]
Changed URL event listener in onMount, to fix build errors I was experiencing. 

Concern: Is it an issue that all PropListing Components have an event listener for the hash changes?

**Comment by kwongz** [comment]
Event listener logic handled in context='module', created and removed once per page

**Comment by kwongz** [comment]
fix #2119

File: sites/docs/components/PropListing.svelte [hunk 1]
```diff
@@ -18,22 +60,47 @@
...
 		} catch (err) {}
 	}
+
+	let counter = 0;
+	const updateIdName = () => {
+		idName = name.replace(/[^a-zA-Z0-9]/g, '-');
+		while (names.has(idName)) {
+			counter++;
+			idName = counter > 1 ? `${name}-${counter}` : `${name}`;
+		}
```
Comment (ItsMeBrianD) [inline sites/docs/components/PropListing.svelte:70]: Is this going to be stable when multiple of the same component appear?

This looks good to me, but is probably a good candidate for unit testing though

**Review by ItsMeBrianD — APPROVED** [review]
lgtm assuming the id / indexing here is stable

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]
Changeset

**Review by ItsMeBrianD — APPROVED** [review]

**Pull request merged**